### PR TITLE
Feature/zen 26864 cdk improvements

### PR DESCRIPTION
**Pull request by mayathegreat**
Adding logic for graph view to make refreshing more visible / less buggy.

### CODE REVIEW

**Comment by mayathegreat** [comment]
retest this

**Comment by mayathegreat** [comment]
screen-before
screen-after
See screenshots above -- note that the toolbar is grayed out when we are refreshing the graphs. Not quite visible but the gear icon is also spinning to indicate it is loading in

File: central-query/src/main/js/visualization/src/Chart.js [hunk 1]
```diff
@@ -658,6 +659,18 @@
...
+        hasPendingRequests: function(){
+            return this.updatePromise && this.updatePromise.state() == "pending";
+        },
+        cancelUpdate: function() {
+            // cancel ajax request (async req)
+            this.updateRequest.abort();
+            this.cleanupDataReq();
+        },
+        cleanupDataReq: function() {
+            clearInterval(this.updateTimeout);
```
Comment (mozart27) [inline central-query/src/main/js/visualization/src/Chart.js]: Does it matter that you're using a `clearInterval()` on a timeout created by `setTimeout()`? This is a question from my ignorance of JavaScript stuff...

File: central-query/src/main/resources/api/visualization.js [hunk 2]
```diff
@@ -1926,6 +1927,18 @@ if (typeof exports !== 'undefined') {
...
+        hasPendingRequests: function(){
+            return this.updatePromise && this.updatePromise.state() == "pending";
+        },
+        cancelUpdate: function() {
+            // cancel ajax request (async req)
+            this.updateRequest.abort();
+            this.cleanupDataReq();
+        },
+        cleanupDataReq: function() {
+            clearInterval(this.updateTimeout);
```
Comment (mozart27) [inline central-query/src/main/resources/api/visualization.js]: Same Q as above - is it ok to use `clearInterval()` when the timeout was created with `setTimeout()` ?

File: central-query/src/main/js/visualization/src/Chart.js (hunk 1 above)
Comment (mayathegreat) [inline central-query/src/main/js/visualization/src/Chart.js]: oooooh good catch! I'd fixed this in an earlier revision that somehow got overwritten. Will make the change

File: central-query/src/main/resources/api/visualization.js (hunk 2 above)
Comment (mayathegreat) [inline central-query/src/main/resources/api/visualization.js]: ditto

**Pull request merged**